community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/StatisticsBackedCardinalityModel.scala

```diff
@@ -42,6 +43,10 @@ class StatisticsBackedCardinalityModel(queryGraphCardinalityModel: QueryGraphCar
     case RegularQueryProjection(_, QueryShuffle(_, None, Some(limit: IntegerLiteral))) =>
       Cardinality.min(in, limit.value.toDouble)
 
+    // Normal projection with LIMIT
+    case RegularQueryProjection(_, QueryShuffle(_, None, Some(unknownLimit))) =>
+      Cardinality.min(in, GraphStatistics.DEFAULT_LIMIT_CARDINALITY)
+
```

Comment:
I was wondering since those limit constants will potentially change the plan I see few value in auto-parametrize them away. It is true that we will spend more time compiling queries, but on the other hand you might compile a plan for `LIMIT 75` with awesome performance, then you cache it and reuse it for `LIMIT 1000` and it performs horribly. (I just made up numbers here).

My understanding of auto-parametrization is to take away everything which is not relevant for planning in order to cache the most general plan to be reused. Parametrizing away constants that effect planning will break the above assumption. No?

Perhaps this is not super important since as you said if users use params they will not be affected by this.